### ISSUE TITLE
Error in Python code generated on Flymine website while accessing templates.

### ISSUE THREAD

**Issue by Sanat-Mishra** [intermine/intermine#1981]
This was the specific template I was accessing.
![screenshot 2019-01-04 at 8 18 21 pm](https://user-images.githubusercontent.com/19981230/50694014-bc156600-105e-11e9-8e30-bb4746c430cd.jpg)

The Python code generated.
![screenshot 2019-01-04 at 8 18 29 pm](https://user-images.githubusercontent.com/19981230/50694050-d94a3480-105e-11e9-8838-6209b025f4ea.jpg)

The syntax error I got while trying to execute on Jupyter. Notice constraint D, the multiple values have been enclosed in "...".
![screenshot 2019-01-04 at 8 18 50 pm](https://user-images.githubusercontent.com/19981230/50694183-32b26380-105f-11e9-89a6-85c0fe28f506.jpg)

The code for it- 
```
# The following two lines will be needed in every python script:
from intermine.webservice import Service
service = Service("http://www.flymine.org/flymine/service")

# Show all the genes known to be involved in a specific KEGG, REACTOME or
# FlyReactome pathway and their orthologous genes in one specific organism.
# (Data Source: KEGG, Reactome, FlyReactome, inParanoid, Drosophila consortium).

template = service.get_template('Pathway_GenesOrthol_new2')

# You can edit the constraint values below
# A    Pathway.name    Show all genes for the following pathway:
# B    Pathway.genes.organism.name    For organism:
# D    Pathway.genes.homologues.homologue.organism.name    and give the orthologous genes from:
# C    Pathway.dataSets.name    Use pathway dataset (KEGG, Reactome or FlyReactome):

rows = template.rows(
    A = {"op": "=", "value": "Pentose phosphate pathway"},
    B = {"op": "=", "value": "Drosophila melanogaster"},
    D = {"op": "ONE OF", "values": "["Anopheles gambiae", "Caenorhabditis elegans", "Drosophila ananassae"]"},
    C = {"op": "=", "value": "KEGG pathways data set"}
)
for row in rows:
    print(row["genes.secondaryIdentifier"], row["genes.symbol"], \
        row["genes.homologues.homologue.secondaryIdentifier"], \
        row["genes.homologues.homologue.symbol"], row["identifier"], row["name"])

```
Finally, on removing the quotes accompanying constraint D, it ran smoothly.
![screenshot 2019-01-04 at 8 19 56 pm](https://user-images.githubusercontent.com/19981230/50694446-f03d5680-105f-11e9-9d64-8d82da554e78.jpg)

Is this an issue pertaining to Python 3 and worked fine on Python 2?

**Comment by tejaswarathe** [comment]
Hi, Is this issue still open to working on?

**Comment by julie-sullivan** [comment]
@tejaswarathe Yes it is! Here is the code: [WebservicePythonCodeGenerator.java](https://github.com/intermine/intermine/blob/dev/intermine/api/src/main/java/org/intermine/api/query/codegen/WebservicePythonCodeGenerator.java)

There is also a corresponding unit test that should be updated.

Thanks! :)

**Comment by Monal5031** [comment]
Its should be an issue on both python versions @Sanat-Mishra , values having multiple constraints should to be a list rather than a single string (I don't know much about the project but I guess that's what I could make of that java code)
Exact line of bug is [here](https://github.com/intermine/intermine/blob/dev/intermine/api/src/main/java/org/intermine/api/query/codegen/WebservicePythonCodeGenerator.java#L569)

**Comment by mbasil09** [comment]
I am gettin an error on 
template = service.get_template('Pathway_GenesOrthol_new2')

Traceback (most recent call last):
  File "C:\Users\mbasi\AppData\Local\Programs\Python\Python37-32\lib\site-packages\intermine\query.py", line 478, in group
    yield tuple([next(itr) for i in range(count)])
  File "C:\Users\mbasi\AppData\Local\Programs\Python\Python37-32\lib\site-packages\intermine\query.py", line 478, in <listcomp>
    yield tuple([next(itr) for i in range(count)])
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\mbasi\AppData\Local\Programs\Python\Python37-32\lib\site-packages\intermine\webservice.py", line 447, in get_template
    t = Template.from_xml(t, self.model, self)
  File "C:\Users\mbasi\AppData\Local\Programs\Python\Python37-32\lib\site-packages\intermine\query.py", line 487, in from_xml
    for path, direction in group(sos, 2):
RuntimeError: generator raised StopIteration

what could be causing this?

**Comment by yochannah** [comment]
@mbasil09 sorry for the slow reply on this :) is this still an issue?

**Comment by mbasil09** [comment]
@yochannah 
Can you try to reproduce this problem?
I mean:
Go to flymine website, and try to generate a python template where you specifically use "ONE OF" or "NONE OF" like the one in this issue has been done.

For your convinience this is the exact link : https://www.flymine.org/flymine/template.do?name=Pathway_GenesOrthol_new2&scope=all

![image](https://user-images.githubusercontent.com/55938020/71520355-a9d0af80-28e1-11ea-9f33-006247c4c5dc.png)

Just change the "=" to "ONE OF" in the third row.

**Comment by yochannah** [comment]
the error above has been transferred to #2171, and the main issue this ticket was addressed was fixed by #2116. 🚀
